testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java

```diff
@@ -67,35 +67,245 @@ public void testColumnMappingModeName()
+
+        try {
+            onDelta().executeQuery("INSERT INTO default." + tableName + " VALUES (0), (9)");
+
+            List<Row> expectedRows = ImmutableList.of(row(0), row(9));
+
+            assertThat(onDelta().executeQuery("SELECT * FROM default." + tableName))
+                    .containsOnly(expectedRows);
+            assertThat(onTrino().executeQuery("SELECT * FROM delta.default." + tableName))
+                    .containsOnly(expectedRows);
+
+            assertThat(onTrino().executeQuery("SHOW STATS FOR delta.default." + tableName))
+                    .containsOnly(ImmutableList.of(
+                            row("mixed_case", null, null, 0.0, null, "0", "9"),
+                            row(null, null, null, null, 2.0, null, null)));
+        }
+        finally {
+            onTrino().executeQuery("DROP TABLE delta.default." + tableName);
+        }
+    }
+
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, PROFILE_SPECIFIC_TESTS})
+    public void testShowStatsFromJsonForColumnMappingModeName()
+    {
+        String tableName = "test_dl_show_stats_for_column_mapping_mode_" + randomTableSuffix();
+
+        onDelta().executeQuery("" +
+                "CREATE TABLE default." + tableName +
+                " (a_number INT)" +
+                " USING delta " +
+                " LOCATION 's3://" + bucketName + "/databricks-compatibility-test-" + tableName + "'" +
+                " TBLPROPERTIES (" +
+                " 'delta.columnMapping.mode' = 'name'," +
+                " 'delta.minReaderVersion' = '2'," +
+                " 'delta.minWriterVersion' = '5')");
+
+        try {
+            onDelta().executeQuery("INSERT INTO default." + tableName + " VALUES (1), (2), (null)");
+
+            assertThat(onTrino().executeQuery("SHOW STATS FOR delta.default." + tableName))
+                    .containsOnly(ImmutableList.of(
+                            row("a_number", null, null, 0.33333333333, null, "1", "2"),
+                            row(null, null, null, null, 3.0, null, null)));
+        }
+        finally {
+            onTrino().executeQuery("DROP TABLE delta.default." + tableName);
+        }
+    }
+
+    @Test(groups = {DELTA_LAKE_DATABRICKS, PROFILE_SPECIFIC_TESTS})
+    public void testShowStatsFromParquetForColumnMappingModeName()
+    {
+        String tableName = "test_dl_show_parquet_stats_for_column_mapping_mode_" + randomTableSuffix();
+
         onDelta().executeQuery("" +
                 "CREATE TABLE default." + tableName +
                 " (a_number INT)" +
                 " USING delta " +
                 " LOCATION 's3://" + bucketName + "/databricks-compatibility-test-" + tableName + "'" +
                 " TBLPROPERTIES (" +
+                " 'delta.columnMapping.mode' = 'name'," +
+                " 'delta.minReaderVersion' = '2'," +
+                " 'delta.minWriterVersion' = '5')");
+
+        try {
+            for (int value = 0; value < 10; value++) {
```

Comment:
There is a config property for tuning checkpoint frequency (it's not guaranteed to be exactly 10).
Use this (and also reduce number of inserts)